client/components/replacePost/index.jsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-underscore-dangle */
+/* global React */
+
+import apiFetch from '@wordpress/api-fetch';
+import { Spinner } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import PostSelector from '../postSelector';
+
+/**
+ * Slotfill to add clone & replace controls to the sidebar.
+ */
+const ReplacePost = () => {
+  const { editPost } = useDispatch('core/editor');
+  const {
+    meta,
+    postType,
+    currentPost,
+  } = useSelect((select) => ({
+    meta: select('core/editor').getEditedPostAttribute('meta') || {},
+    postType: select('core/editor').getCurrentPostType(),
+    currentPost: select('core/editor').getCurrentPost(),
+  }), []);
+
+  const [replacePostId, setReplacePostId] = useState(meta._cr_replace_post_id);
+  const [replacePost, setReplacePost] = useState(false);
+  const selected = replacePost ? [replacePost] : [];
+
+  const fetchPost = async (postId) => {
+    const post = await apiFetch({ path: `/wp/v2/${postType}/${postId}` });
```

Comment:
We might be able to use `getEntityRecord` here to have Gutenberg be responsible for the data fetch instead of doing it manually, but this is also fine.